### PR TITLE
Add Configuration Types

### LINKED ISSUE #36: Update Conditions and Validators to Use Configs
### Description

Many of the conditions and validators are configurable. These configurations should be condensed into `struct`'s with the appropriate default values in the initializer. Then the config can be passed into the condition or validator rather than a whole slew of arguments. This will allow for easier extension in the future and keep method names short.

### PR DESCRIPTION
**Pull request by ghost**
Resolves #36 `Update Conditions and Validators to Use Configs`.

### Summary

Adds configuration types for conditions and validators as appropriate.

### Details

- Adds Configuration protocol
- Adds ConfigurableCondition and ConfigurableValidator protocol
- Updates Alphabetic, Alphanumeric, CreditCard, PasswordStrength, Postcode, and Range conditions and validators
- Update Gem dependencies

### CODE REVIEW

File: Sources/Configurations/AlphabeticConfiguration.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public let allowsWhitespace: Bool
+    
+    
+    // MARK: - Initializers
+    
+    public init() {
+        self.init(allowsUnicode: false, allowsWhitespace: false)
+    }
+    
+    public init(allowsUnicode: Bool = false, allowsWhitespace: Bool = false) {
```
Comment (madhikarma) [inline Sources/Configurations/AlphabeticConfiguration.swift:30]: What did you think about using the struct's default parameter values before deciding on an initializer based approach? Is it about helping enforce an initialization for all configurations?

File: Sources/Configurations/PasswordStrengthConfiguration.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//  PasswordStrengthConfiguration.swift
+//  FormValidatorSwift
+//
+//  Created by Aaron McTavish on 03/01/2017.
+//  Copyright © 2017 ustwo. All rights reserved.
+//
+
+import Foundation
+
+
```
Comment (madhikarma) [inline Sources/Configurations/PasswordStrengthConfiguration.swift:11]: Documentation?

File: Tests/Unit Tests/Configurations/ConfigurationSeeds.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+//
+//  Created by Aaron McTavish on 03/01/2017.
+//  Copyright © 2017 ustwo. All rights reserved.
+//
+
+import Foundation
+
+@testable import FormValidatorSwift
+
+
```
Comment (madhikarma) [inline Tests/Unit Tests/Configurations/ConfigurationSeeds.swift:13]: Documentation? Just in case people aren't familiar with seed terminology

**Review by madhikarma — APPROVED** [review]

**Comment by madhikarma** [comment]
Looks great @aamctustwo Just a question and some small suggestions. Nice and tidy work!

File: Sources/Configurations/AlphabeticConfiguration.swift (hunk 1 above)
Comment (madhikarma) [reply Sources/Configurations/AlphabeticConfiguration.swift:30]: Ignore. As discussed with @aamctustwo the struct has non-mutable properties but we want to be able to configure them and give them default hence the initializer

File: Sources/Configurations/PasswordStrengthConfiguration.swift (hunk 2 above)
Comment (ghost) [reply Sources/Configurations/PasswordStrengthConfiguration.swift:11]: Done

File: Tests/Unit Tests/Configurations/ConfigurationSeeds.swift (hunk 3 above)
Comment (ghost) [reply Tests/Unit Tests/Configurations/ConfigurationSeeds.swift:13]: Done

File: Sources/Configurations/CreditCardConfiguration.swift [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        add(description: "VISA", forCardType: CreditCardType.visa, toResultString: &result)
+        
+        return result
+    }
+    
+    /**
+     Combined regular expression string for set of credit cards.
+     */
+    public var regex: String {
+        var result = ""
```
Comment (madhikarma) [inline Sources/Configurations/CreditCardConfiguration.swift:52]: Do you want to newline before and after `result`?

File: Sources/Configurations/CreditCardConfiguration.swift [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            resultString += descriptionString
+        }
+    }
+    
+    /**
+     Regular expression values for credit card types.
+     */
+    private var cardRegex: String? {
+        switch self {
+        case CreditCardType.americanExpress:
```
Comment (madhikarma) [inline Sources/Configurations/CreditCardConfiguration.swift:80]: I don't suppose we have any links to documentation or reference we can add here for the regex e.g. Luhn algorithm?

File: Sources/Configurations/CreditCardConfiguration.swift [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    static let allArray: [CreditCardType] = [.americanExpress, .dinersClub, .discover, .jcb, .maestro, .mastercard, .visa]
+    
+    /**
+     Literal description for set of credit cards, separated with commas and single space.
+     */
+    public var description: String {
+        
+        var result = ""
+        
+        add(description: "American Express", forCardType: CreditCardType.americanExpress, toResultString: &result)
```
Comment (madhikarma) [inline Sources/Configurations/CreditCardConfiguration.swift:37]: Do you think it's worth localizing these? Or are they debug descriptions (though maybe even so)

**Review by madhikarma — APPROVED** [review]
Final changes look good @aamctustwo just some small moment or suggestion to consider

File: Sources/Configurations/CreditCardConfiguration.swift (hunk 6 above)
Comment (ghost) [reply Sources/Configurations/CreditCardConfiguration.swift:37]: May or may not be worth it. Many languages do not localise company names. Also, this was just moving the existing OptionSet from the Condition to the new Configuration.

File: Sources/Configurations/CreditCardConfiguration.swift (hunk 4 above)
Comment (ghost) [reply Sources/Configurations/CreditCardConfiguration.swift:52]: Cleaned.

**Pull request merged**